File: backend/src/database/track.py
from ..addons import db

class Track(db.Model):
    id      = db.Column(db.Integer, primary_key=True)
    
    isrc    = db.Column(db.String, unique=True)
    title   = db.Column(db.String)
    artists = db.Column(db.String)
    album   = db.Column(db.String)
    release = db.Column(db.DateTime)

    youtube = db.Column(db.String, nullable=True)
    spotify = db.Column(db.String, nullable=True)
    deezer  = db.Column(db.Integer, nullable=True)

File: frontend/src/app/types/track.ts
import { Album } from './album';
import { Artist } from './artist';

export interface Identifier {
  spotify: string;
  youtube?: string;
  deezer?: string;
}

export interface Track {
  isrc: string;
  title: string;
  artist: Artist['id'];
  album: Album['id'];
  release: string;
  external_ids: Identifier;
}
